pkg/settings/settings_helper.go

```diff
@@ -168,13 +166,38 @@ type ImagePreloadOption struct {
 	Strategy PreloadStrategy `json:"strategy,omitempty"`
 }
 
+type ModeType string
+
+const (
+	// Every node enters the pre-draining state automatically (default mode)
+	AutoType ModeType = "auto"
+	// Each node waits indefinitely before entering the pre-draining state until the user agrees to continue
+	ManualType ModeType = "manual"
+)
+
+type NodeUpgradeStrategy struct {
+	Mode *ModeType `json:"mode,omitempty"`
+
+	// PauseNodes contains the node names that should be paused before entering the pre-draining state during the node
+	// upgrade phase. It is only honored when the Mode is "manual". If the Mode is "manual" but no node names are
+	// provided, all nodes will be paused before entering the pre-draining state.
+	PauseNodes []string `json:"pauseNodes,omitempty"`
```

Comment:
what do you think of using node labels to allow more flexibility with grouping and selections? for per node identification, user can also use the `kubernetes.io/hostname` label.